lib/ladybug/energy_model/schedule_ruleset.rb

```diff
@@ -0,0 +1,134 @@
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class ScheduleRuleset < ModelObject
+      attr_reader :errors, :warnings
+            
+      def initialize(hash = {})
+        super(hash)
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getScheduleRulesetByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_schedule_ruleset = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+        openstudio_schedule_ruleset.setName(@hash[:name])
+        schedule_type_limits = OpenStudio::Model::ScheduleTypeLimits.new(openstudio_model)
+        schedule_type_limits.setUnitType(@hash[:schedule_type_limits][:unit_type])
+        schedule_type_limits.setLowerLimitValue(@hash[:schedule_type_limits][:lower_limit_value])
+        schedule_type_limits.setUpperLimitValue(@hash[:schedule_type_limits][:upper_limit_value])
+        case 
+        when @hash[:schedule_type_limits][:numeric_type][:type] == 'ScheduleContinuous'
+          schedule_type_limits.setNumericType('Continuous')
+        when @hash[:schedule_type_limits][:numeric_type][:type] == 'ScheduleDiscrete'
+          schedule_type_limits.setNumericType('Discrete')
+        end
+        openstudio_summer_day = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+        openstudio_schedule_ruleset.setSummerDesignDaySchedule(openstudio_summer_day)
+        openstudio_schedule_ruleset.summerDesignDaySchedule.setName(@hash[:summer_designday_schedule][:name])
```

Comment:
Would save some line space to just call `openstudio_summer_day.setName(@hash[:summer_designday_schedule][:name])`, the objects are linked so you don't have to get it from the openstudio_schedule_ruleset each time